src/org/exist/client/InteractiveClient.java

```diff
@@ -1309,18 +1346,15 @@ private synchronized boolean findRecursive(final Collection collection, final Pa
     /**
      * stores given Resource
      *
-     * @param fileName simple file or directory
+     * @param file file or directory
      * @throws XMLDBException
      */
-    protected synchronized boolean parse(String fileName) throws XMLDBException {
+    protected synchronized boolean parse(final Path file) throws XMLDBException {
```

Comment:
I think you misunderstand. `java.nio.file.Paths` and `java.nio.file.Path` are platform aware. So if you run your own test code (from above) on Windows, you will get different output (which is correct for Windows):

```
test/test.file > old > test\test.file
test/test.file > new > test\test.file
file name = test.file
test\test.file > old > test\test.file
test\test.file > new > test\test.file
file name = test.file
```

Which is why I told you that testing my code (which I did on both Mac and Windows) produces the correct results.